src/bluesearch/entrypoint/database/topic_filter.py

```diff
@@ -0,0 +1,188 @@
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+"""Filter articles with relevant topics."""
+from __future__ import annotations
+
+import argparse
+import logging
+import typing
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+from bluesearch.database.topic_info import TopicInfo
+from bluesearch.database.topic_rule import TopicRule, check_accepted
+
+logger = logging.getLogger(__name__)
+
+
+def init_parser(parser: argparse.ArgumentParser) -> argparse.ArgumentParser:
+    """Initialise the argument parser for the topic-filter subcommand.
+
+    Parameters
+    ----------
+    parser
+        The argument parser to initialise.
+
+    Returns
+    -------
+    argparse.ArgumentParser
+        The initialised argument parser. The same object as the `parser`
+        argument.
+    """
+    parser.description = "Filter articles with relevant topics"
+
+    parser.add_argument(
+        "extracted_topics",
+        type=Path,
+        help="""
+        Path to a .JSONL file that was an output of the `topic-extract`
+        command.
+        """,
+    )
+    parser.add_argument(
+        "filter_config",
+        type=Path,
+        help="""
+        Path to a .JSONL file that defines all the rules for filtering.
+        """,
+    )
+    parser.add_argument(
+        "output_file",
+        type=Path,
+        help="""
+        Path to a .CSV file where rows are different articles
+        and columns contain relevant information about these articles.
+        """,
+    )
+
+    return parser
+
+
+def parse_filter_config(config: list[dict]) -> tuple[list[TopicRule], list[TopicRule]]:
+    """Parse filter configuration.
+
+    Parameters
+    ----------
+    config
+        Topic Rules configuration
+
+    Returns
+    -------
+    topic_rules_accept : list[TopicRule]
+        List of accepted TopicRule
+    topic_rules_reject : list[TopicRule]
+        List of rejected TopicRule
```

Comment:
```suggestion
        List of acceptance TopicRule
    topic_rules_reject : list[TopicRule]
        List of rejection TopicRule
```
See similar comment in the docstring below.